File: AspireIssueTracker.Issues.Data/Issues/Issue.cs
using AspireIssueTracker.Issues.Data.Users;
using MongoDB.Bson;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AspireIssueTracker.Issues.Data.Issues
{
	public record Issue
	{
		public ObjectId Id { get; init; }

		public required string Title { get; init; }
		public string Description { get; init; } = "";
		public IssueStatus Status { get; init; }
		public IssuePriority Priority { get; init; }
		public User? AssignedTo { get; init; }
		public User? ReportedBy { get; init; }
		public DateTime CreatedAt { get; init; }
		public DateTime UpdatedAt { get; init; }
		public List<IssueComment> Comments { get; init; } = [];
	}

	public enum IssueStatus
	{
		Open,
		InProgress,
		Rejected,
		Resolved,
		Closed
	}

	public enum IssuePriority
	{
		Low,
		Medium,
		High,
		Critical
	}

	// Potential future enhancement
	public enum IssueType
	{
		Bug,
		Feature,
		Task,
		Improvement
	}

	public record IssueComment
	{
		public required string Id { get; init; }

		public int SortOrder { get; init; }
		public string Comment { get; init; } = "";
		public User? CommentedBy { get; init; }
		public DateTime CommentedAt { get; init; }
		public DateTime UpdatedAt { get; init; }
	}
}

File: AspireIssueTracker.UI/src/types/Issue.ts
export enum IssueStatus {
  open,
  inProgress,
  rejected,
  resolved,
  closed,
}

export enum IssuePriority {
  low,
  medium,
  high,
  critical,
}

export type Person = {
  id: string;
  name: string;
};

export type IssueComment = {
  id: string;
  sortOrder: number;
  comment: string;
  commentedBy: Person;
  commentedAt: Date;
  updatedAt: Date;
};

export type Issue = {
  id: string;
  title: string;
  description: string;
  status: IssueStatus;
  priority: IssuePriority;
  assignedTo?: Person;
  reportedBy?: Person;
  createdAt: Date;
  updatedAt: Date;
  comments: IssueComment[];
};

export const issuePriorityMap: Record<IssuePriority, string> = {
  [IssuePriority.low]: "Low",
  [IssuePriority.medium]: "Medium",
  [IssuePriority.high]: "High",
  [IssuePriority.critical]: "Critical",
};

export const issueStatusMap: Record<IssueStatus, string> = {
  [IssueStatus.open]: "Open",
  [IssueStatus.inProgress]: "In Progress",
  [IssueStatus.rejected]: "Rejected",
  [IssueStatus.resolved]: "Resolved",
  [IssueStatus.closed]: "Closed",
};
